Library/SharedFunctionsTests.swift

```diff
@@ -182,4 +182,62 @@ final class SharedFunctionsTests: TestCase {
     XCTAssertTrue(ksr_isOSVersionAvailable(12.123))
     XCTAssertTrue(ksr_isOSVersionAvailable(12.9))
   }
+
+  func testAttributedStringWithLinks() {
+    let string = "What a lovely string with a link and another link" as NSString
+    let attrString = NSAttributedString(string: string as String)
+    let links: [AttributedLinkData] = [
+      ("a link", URL(string: "link://a_link"), [.foregroundColor: UIColor.blue]),
+      ("another link", URL(string: "link://another_link"), [:])
+    ]
+    let stringWithLinks = ksr_attributedString(attrString, with: links)
+
+    let aLinkRange = string.range(of: "a link")
+    let anotherLinkRange = string.range(of: "another link")
+    let fullRange = string.range(of: string as String)
+
+    let aLinkAttribute = stringWithLinks
+      .attribute(.link, at: aLinkRange.location, longestEffectiveRange: nil, in: fullRange)
+    let aLinkColorAttribute = stringWithLinks
+      .attribute(.foregroundColor, at: aLinkRange.location, longestEffectiveRange: nil, in: fullRange)
+    let anotherLinkAttribute = stringWithLinks
+      .attribute(.link, at: anotherLinkRange.location, longestEffectiveRange: nil, in: fullRange)
+    let anotherLinkColorAttribute = stringWithLinks
+      .attribute(.foregroundColor, at: anotherLinkRange.location, longestEffectiveRange: nil, in: fullRange)
+
+    XCTAssertNotNil(aLinkAttribute, "a link is a link")
+    XCTAssertNotNil(aLinkColorAttribute, "a link has a foreground color")
+    XCTAssertNotNil(anotherLinkAttribute, "another link is a link")
+    XCTAssertNil(anotherLinkColorAttribute, "another link has no foreground color")
+  }
+
+  func testByPledgingYouAgree() {
+    let string = Strings.By_pledging_you_agree() as NSString
```

Comment:
Something worth noting is that this string, _By_pledging_you_agree_ is actually the following text:

> By pledging you agree to Kickstarter’s Terms of Use, Privacy Policy and Cookie Policy.

In our designs these are links:

<img width="198" alt="Screen Shot 2019-05-07 at 9 24 00 PM" src="https://user-images.githubusercontent.com/3735375/57349313-77820000-710e-11e9-86e6-0b1e459c0e85.png">

This _may_ be problematic when it comes to translations but let's chat more offline.